benches/benches/computepass.rs

```diff
@@ -126,22 +126,14 @@ impl ComputepassState {
 
         let mut texture_views = Vec::with_capacity(texture_count);
         for i in 0..texture_count {
-            let texture = device_state
-                .device
-                .create_texture(&wgpu::TextureDescriptor {
-                    label: Some(&format!("Texture {i}")),
-                    size: wgpu::Extent3d {
-                        width: 1,
-                        height: 1,
-                        depth_or_array_layers: 1,
-                    },
-                    mip_level_count: 1,
-                    sample_count: 1,
-                    dimension: wgpu::TextureDimension::D2,
-                    format: wgpu::TextureFormat::Rgba8UnormSrgb,
-                    usage: wgpu::TextureUsages::TEXTURE_BINDING,
-                    view_formats: &[],
-                });
+            let texture = device_state.device.create_texture(
+                &wgpu::TextureDescriptor::builder()
```

Comment:
Hey @ErichDonGubler, I've just implemented support for the syntax above which allows you to specify with `#[builder(pos = start_fn/finish_fn)]` that the parameter should be passed as a positional parameter of the `start_fn` or `finish_fn`. https://github.com/elastio/bon/pull/125

Note that I haven't tested this feature exhaustively yet, but I [tested some simple cases](https://github.com/elastio/bon/blob/1eea830803dabb220ff1988cb9e48d7bd98a0aa6/bon/tests/integration/builder/positional_members.rs#L19-L145). I'll continue testing it and adding documentation for this feature. Hopefully, by the end of this week this will be released in a new minor release.

If you'd like to try this out, you can add a dependency on the dev version of `bon` from my branch like this:

```rust
bon = { git = "https://github.com/elastio/bon", branch = "feat/positional-only-members" }
```

Reply:
I'm not sure how familiar you are with WebGPU and WGPU, and their historical context is pretty significant here. I'll try to explain briefly, but LMK if something doesn't make immediate sense to you.

1. We have a [JS analogue](https://www.w3.org/TR/webgpu/) that we're representing here. We'd like calling code to `wgpu` to look more-or-less like the corresponding JS, unless we have a _really_ good reason to diverge.
2. Having consistent Rust analogues of JS API analogues is one of the most important tools we have for building a [“pit of success”](https://blog.codinghorror.com/falling-into-the-pit-of-success/) for WGPU users. Many of them come to use WGPU without having significant experience in Rust. This minimizes the cognitive load of the extremely broad API that we have; if you know how to construct one descriptor, then you've already ascended the learning curve for the rest (minus domain knowledge of what you want _in_ them, OFC 🙂).
3. We also have _a lot_ of existing users.

(1) drove the current shape of the API, and as you've already guessed, (3) means that we need backwards-compatible builder APIs if we want to tractably ship. This effectively forces the first iteration that we ship to be a separate (feature-gated) API in `wgpu`, or even a separate crate outside `wgpu`.

All this to say: I can't/won't accept a suggestion to _replace_ existing APIs, but I might consider a builder API generated from _separate_ method definitions. If we do, though, then that needs to be available for _all_ APIs, and not just ones with required fields. That's a lot of effort I'm not willing to go to, yet, but it'd be interesting to add to this current experiment, I think!

Reply:
I've updated the OP to express the openness to trying method builders.